File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/RecipientsService.java
package com.redsun.service.service;

import com.redsun.service.entities.Recipients;
import com.redsun.service.entities.Result;

/**
 * Recipients Service interface
 */
public interface RecipientsService {
	
    // Insert.
	Result insert(final Recipients recipients) ;

    // Update.
	Result update(final Recipients recipients) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listRecipientsForPageAndFilter(final int itemsPerPage, final int pageNo, final Recipients recipients);

}

File: 5.implementation/NXP/redsun/src/main/webapp/static/js/services/RecipientsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('recipientService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(recipient) {
			var urlServer = $rootScope.makePostURL("/recipient/insert");
			return $http.post(urlServer, recipient);
		}
		
		// Update.
		, update: function(recipientId, recipient) {
			var urlServer = $rootScope.makePostURL("/recipient/update/" + recipientId);
			return $http.put(urlServer, recipient);
		}
		
		// Delete.
		, delete: function(recipientId) {
			var urlServer = $rootScope.makePostURL("/recipient/delete/" + recipientId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(recipientId) {
			var url = $rootScope.makeGetURL("/recipient/getbyid/" + recipientId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, recipient) {
			var url = $rootScope.makeGetURL("/recipient/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: recipient });
		}
		
	}// return.
});